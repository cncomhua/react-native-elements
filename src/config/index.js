import BackgroundImage from './BackgroundImage';
import colors from './colors';
import ThemeProvider from './ThemeProvider';
import getTheme from './getTheme';
import ViewPropTypes from './ViewPropTypes';
import fonts from './fonts';

export {
  BackgroundImage,
  colors,
  fonts,
  ThemeProvider,
  getTheme,
  ViewPropTypes,
};
