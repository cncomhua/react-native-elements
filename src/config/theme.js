// For testing purposes only
import colors from './colors';

export default {
  colors,
};
