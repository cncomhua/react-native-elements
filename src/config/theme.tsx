import { ViewProps } from 'react-native';
import { SwipeRatingProps } from '../Rating';
import { TapRatingProps } from '../AirbnbRating';
import { AvatarProps, AccessoryProps } from '../Avatar';
import { BadgeProps } from '../Badge';
import { BottomSheetProps } from '../BottomSheet';
import { ButtonProps } from '../Button';
import { ButtonGroupProps } from '../ButtonGroup';
import { CardProps } from '../card/Card';
import { CheckBoxProps } from '../Checkbox';
import { DividerProps } from '../Divider';
import { HeaderProps } from '../Header';
import { IconProps } from '../Icon';
import { ImageProps } from '../Image';
import { InputProps } from '../Input';
import { ListItemProps } from '../ListItem';
import { ListItemAccordionProps } from '../ListItem';
import { OverlayProps } from '../Overlay';
import { PricingCardProps } from '../PricingCard';
import { SearchBarProps } from '../searchbar/SearchBar';
import { SliderProps } from '../Slider';
import { SocialIconProps } from '../SocialIcon';
import { TextProps } from '../Text';
import { TileProps } from '../Tile';
import { TooltipProps } from '../Tooltip';
import { SwitchProps } from '../Switch';
import { TabItemProps, TabProps } from '../tab/Tab';
import { TabViewProps } from '../tab/TabView';
import { FABProps } from '../FAB';
import { SpeedDialProps, SpeedDialActionProps } from '../SpeedDial';
import { LinearProgressProps } from '../LinearProgress';
import { ChipProps } from '../Chip';
import {
  DialogActionsProps,
  DialogLoadingProps,
  DialogProps,
  DialogTitleProps,
} from '../Dialog';

import colors, { Colors } from './colors';

export default {
  colors,
};

type RecursivePartial<T> = { [P in keyof T]?: RecursivePartial<T[P]> };

export interface FullTheme {
  Avatar: Partial<AvatarProps>;
  AvatarAccessory: Partial<AccessoryProps>;
  Badge: Partial<BadgeProps>;
  BottomSheet: Partial<BottomSheetProps>;
  Button: Partial<ButtonProps>;
  ButtonGroup: Partial<ButtonGroupProps>;
  Chip: Partial<ChipProps>;
  Card: Partial<CardProps>;
  CardDivider: Partial<DividerProps>;
  CardFeaturedSubtitle: Partial<TextProps>;
  CardFeaturedTitle: Partial<TextProps>;
  CardImage: Partial<ImageProps>;
  CardTitle: Partial<TextProps>;
  CheckBox: Partial<CheckBoxProps>;
  Divider: Partial<DividerProps>;
  Dialog: Partial<DialogProps>;
  DialogActions: Partial<DialogActionsProps>;
  DialogButton: Partial<ButtonProps>;
  DialogLoading: Partial<DialogLoadingProps>;
  DialogTitle: Partial<DialogTitleProps>;
  Header: Partial<HeaderProps>;
  Icon: Partial<IconProps>;
  Image: Partial<ImageProps>;
  Input: Partial<InputProps>;
  ListItem: Partial<ListItemProps>;
  ListItemAccordion: Partial<ListItemAccordionProps>;
  ListItemButtonGroup: Partial<ButtonGroupProps>;
  ListItemCheckBox: Partial<CheckBoxProps>;
  ListItemContent: Partial<ViewProps>;
  ListItemChevron: Partial<IconProps>;
  ListItemInput: Partial<InputProps>;
  ListItemSubtitle: Partial<TextProps>;
  ListItemTitle: Partial<TextProps>;
  Overlay: Partial<OverlayProps>;
  PricingCard: Partial<PricingCardProps>;
  Rating: Partial<SwipeRatingProps>;
  AirbnbRating: Partial<TapRatingProps>;
  SearchBar: Partial<SearchBarProps>;
  Slider: Partial<SliderProps>;
  SocialIcon: Partial<SocialIconProps>;
  Text: Partial<TextProps>;
  Tile: Partial<TileProps>;
  Switch: Partial<SwitchProps>;
  Tooltip: Partial<TooltipProps>;
  colors: RecursivePartial<Colors>;
  Tab: Partial<TabProps>;
  TabItem: Partial<TabItemProps>;
  TabView: Partial<TabViewProps>;
  TabViewItem: Partial<ViewProps>;
  FAB: Partial<FABProps>;
  SpeedDial: Partial<SpeedDialProps>;
  SpeedDialAction: Partial<SpeedDialActionProps>;
  LinearProgress: Partial<LinearProgressProps>;
}

export type Theme<T = {}> = Partial<FullTheme> & T;

export type UpdateTheme = (updates: RecursivePartial<FullTheme>) => void;

export type ReplaceTheme = (updates: RecursivePartial<FullTheme>) => void;
