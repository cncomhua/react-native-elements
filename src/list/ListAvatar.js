import React from 'react';
import Avatar from '../avatar/Avatar';

function ListAvatar(props) {
  return <Avatar size={40} rounded {...props} />;
}

export default ListAvatar;
