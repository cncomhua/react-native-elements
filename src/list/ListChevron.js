import React from 'react';
import { Platform } from 'react-native';
import Icon from '../icons/Icon';

function ListChevron(props) {
  return (
    <Icon
      name={
        Platform.OS === 'ios' ? 'ios-arrow-forward' : 'keyboard-arrow-right'
      }
      type={Platform.OS === 'ios' ? 'ionicon' : 'material'}
      size={16}
      color="#D1D1D6"
      {...props}
    />
  );
}

export default ListChevron;
