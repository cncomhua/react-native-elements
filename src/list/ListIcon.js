import React from 'react';
import { Platform } from 'react-native';
import Icon from '../icons/Icon';

const ANDROID_SECONDARY = 'rgba(0, 0, 0, 0.54)';

function ListIcon(props) {
  return (
    <Icon
      size={24}
      color={Platform.OS === 'ios' ? null : ANDROID_SECONDARY}
      {...props}
    />
  );
}

export default ListIcon;
