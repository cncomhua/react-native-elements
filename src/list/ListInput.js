import React from 'react';
import { StyleSheet } from 'react-native';
import Input from '../input/Input';

function ListInput({
  inputStyle,
  inputContainerStyle,
  containerStyle,
  ...props
}) {
  return (
    <Input
      renderErrorMessage={false}
      {...props}
      inputStyle={StyleSheet.flatten([styles.input, inputStyle])}
      inputContainerStyle={StyleSheet.flatten([
        styles.inputContentContainer,
        inputContainerStyle,
      ])}
      containerStyle={StyleSheet.flatten([
        styles.inputContainer,
        containerStyle,
      ])}
    />
  );
}

const styles = {
  inputContainer: {
    flex: 1,
    paddingRight: 0,
  },
  inputContentContainer: {
    flex: 1,
    borderBottomWidth: 0,
    width: null,
    height: null,
  },
  input: {
    flex: 1,
    textAlign: 'right',
    width: null,
    height: null,
  },
};

export default ListInput;
