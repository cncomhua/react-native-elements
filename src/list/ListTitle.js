import React from 'react';
import { Platform, StyleSheet, View } from 'react-native';
import Text from '../text/Text';
import { renderNode } from '../helpers';
import { withTheme } from '../config';

const ANDROID_SECONDARY = 'rgba(0, 0, 0, 0.54)';

const renderText = (content, defaultProps, style) =>
  renderNode(Text, content, {
    ...defaultProps,
    style: StyleSheet.flatten([style, defaultProps && defaultProps.style]),
  });

function ListTitle({
  title,
  subtitle,
  titleProps,
  subtitleProps,
  titleStyle,
  subtitleStyle,
  containerStyle,
  alignRight,
  ...props
}) {
  return (
    <View
      style={StyleSheet.flatten([
        styles.contentContainer,
        alignRight && styles.rightContentContainer,
        containerStyle,
      ])}
    >
      {renderText(
        title,
        { testID: 'listItemTitle', ...titleProps },
        StyleSheet.flatten([
          styles.title,
          alignRight && styles.rightTitle,
          titleStyle,
        ])
      )}
      {renderText(
        subtitle,
        subtitleProps,
        StyleSheet.flatten([
          styles.subtitle,
          alignRight && styles.rightSubtitle,
          subtitleStyle,
        ])
      )}
    </View>
  );
}

const styles = {
  contentContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-start',
  },
  title: {
    backgroundColor: 'transparent',
    ...Platform.select({
      ios: {
        fontSize: 17,
      },
      default: {
        fontSize: 16,
      },
    }),
  },
  subtitle: {
    backgroundColor: 'transparent',
    ...Platform.select({
      ios: {
        fontSize: 15,
      },
      default: {
        color: ANDROID_SECONDARY,
        fontSize: 14,
      },
    }),
  },
  rightContentContainer: {
    flex: 0.5,
    alignItems: 'flex-end',
  },
  rightTitle: {
    color: ANDROID_SECONDARY,
  },
  rightSubtitle: {
    color: ANDROID_SECONDARY,
  },
};

export default withTheme(ListTitle, 'ListTitle');
