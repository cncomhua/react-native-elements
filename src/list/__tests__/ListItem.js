import React from 'react';
import { Text, Switch } from 'react-native';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { create } from 'react-test-renderer';

import theme from '../../config/theme';
import { ThemeProvider } from '../../config';

import ThemedListItem, { ListItem } from '../ListItem';
import ListAvatar from '../ListAvatar';
import ListIcon from '../ListIcon';
import ListTitle from '../ListTitle';
import ListChevron from '../ListChevron';
import ListInput from '../ListInput';

describe('ListItem component', () => {
  it('should render without issues', () => {
    const component = shallow(<ListItem theme={theme} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with avatar', () => {
    const component = shallow(
      <ListItem theme={theme} containerStyle={{ backgroundColor: 'peru' }}>
        <ListAvatar source="avatar_uri" />
      </ListItem>
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with left icon', () => {
    const component = shallow(
      <ListItem theme={theme}>
        <ListIcon name="wifi" type="font-awesome" color="red" size={20} />
      </ListItem>
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with Text component', () => {
    const component = shallow(
      <ListItem theme={theme}>
        <Text>Test element</Text>
      </ListItem>
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with title and subtitle', () => {
    const component = shallow(
      <ListItem theme={theme}>
        <ListTitle
          title="title test"
          titleProps={{ numberOfLines: 5 }}
          subtitle="title test"
          titleStyle={{ backgroundColor: 'peru' }}
          subtitleStyle={{ backgroundColor: 'peru' }}
        />
        <ListTitle right title="title" />
      </ListItem>
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with switch', () => {
    const component = shallow(
      <ListItem theme={theme} bottomDivider>
        <Switch value={true} />
        <ListChevron />
      </ListItem>
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with input', () => {
    const component = shallow(
      <ListItem theme={theme}>
        <ListInput placeholder="Enter Text" />
      </ListItem>
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should apply values from theme', () => {
    const testTheme = {
      ListTitle: {
        title: 'List Title',
      },
    };

    const component = create(
      <ThemeProvider theme={testTheme}>
        <ThemedListItem>
          <ListTitle />
        </ThemedListItem>
      </ThemeProvider>
    );

    expect(
      component.root.findByProps({ testID: 'listItemTitle' }).props.children
    ).toBe('List Title');
    expect(component.toJSON()).toMatchSnapshot();
  });
});
