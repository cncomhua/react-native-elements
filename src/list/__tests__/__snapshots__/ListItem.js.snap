// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component should apply values from theme 1`] = `
<View
  replaceTheme={[Function]}
  title=""
  updateTheme={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#bcbbc1",
        "flexDirection": "row",
        "padding": 14,
      }
    }
    testID="padView"
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "fontSize": 17,
          }
        }
        testID="listItemTitle"
        theme={
          Object {
            "ListTitle": Object {
              "title": "List Title",
            },
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
      >
        List Title
      </Text>
    </View>
  </View>
</View>
`;

exports[`ListItem component should render with Text component 1`] = `
<View
  title=""
>
  <PadView
    Component={[Function]}
    pad={16}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#bcbbc1",
        "flexDirection": "row",
        "padding": 14,
      }
    }
  >
    <Text>
      Test element
    </Text>
  </PadView>
</View>
`;

exports[`ListItem component should render with avatar 1`] = `
<View
  title=""
>
  <PadView
    Component={[Function]}
    pad={16}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "peru",
        "borderColor": "#bcbbc1",
        "flexDirection": "row",
        "padding": 14,
      }
    }
  >
    <ListAvatar
      source="avatar_uri"
    />
  </PadView>
</View>
`;

exports[`ListItem component should render with input 1`] = `
<View
  title=""
>
  <PadView
    Component={[Function]}
    pad={16}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#bcbbc1",
        "flexDirection": "row",
        "padding": 14,
      }
    }
  >
    <ListInput
      placeholder="Enter Text"
    />
  </PadView>
</View>
`;

exports[`ListItem component should render with left icon 1`] = `
<View
  title=""
>
  <PadView
    Component={[Function]}
    pad={16}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#bcbbc1",
        "flexDirection": "row",
        "padding": 14,
      }
    }
  >
    <ListIcon
      color="red"
      name="wifi"
      size={20}
      type="font-awesome"
    />
  </PadView>
</View>
`;

exports[`ListItem component should render with switch 1`] = `
<View
  title=""
>
  <PadView
    Component={[Function]}
    pad={16}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomWidth": 0.5,
        "borderColor": "#bcbbc1",
        "flexDirection": "row",
        "padding": 14,
      }
    }
  >
    <Switch
      value={true}
    />
    <ListChevron />
  </PadView>
</View>
`;

exports[`ListItem component should render with title and subtitle 1`] = `
<View
  title=""
>
  <PadView
    Component={[Function]}
    pad={16}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#bcbbc1",
        "flexDirection": "row",
        "padding": 14,
      }
    }
  >
    <Themed.ListTitle
      subtitle="title test"
      subtitleStyle={
        Object {
          "backgroundColor": "peru",
        }
      }
      title="title test"
      titleProps={
        Object {
          "numberOfLines": 5,
        }
      }
      titleStyle={
        Object {
          "backgroundColor": "peru",
        }
      }
    />
    <Themed.ListTitle
      right={true}
      title="title"
    />
  </PadView>
</View>
`;

exports[`ListItem component should render without issues 1`] = `
<View
  title=""
>
  <PadView
    Component={[Function]}
    pad={16}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "#bcbbc1",
        "flexDirection": "row",
        "padding": 14,
      }
    }
  />
</View>
`;
